frame/support/src/storage/generator/double_map.rs

```diff
@@ -431,33 +463,109 @@ impl<
 	}
 }
 
+/// Test iterators for StorageDoubleMap
 #[cfg(test)]
-mod test {
-	use sp_io::TestExternalities;
-	use crate::storage::{self, StorageDoubleMap};
-	use crate::hash::Twox128;
+#[allow(dead_code)]
+mod test_iterators {
+	use codec::{Encode, Decode};
+	use crate::storage::{generator::StorageDoubleMap, IterableStorageDoubleMap, unhashed};
+
+	pub trait Trait {
+		type Origin;
+		type BlockNumber;
+	}
+
+	crate::decl_module! {
+		pub struct Module<T: Trait> for enum Call where origin: T::Origin {}
+	}
+
+	#[derive(PartialEq, Eq, Clone, Encode, Decode)]
+	struct NoDef(u32);
+
+	crate::decl_storage! {
+		trait Store for Module<T: Trait> as Test {
+			DoubleMap: double_map hasher(blake2_128_concat) u16, hasher(blake2_128_concat) u32 => u64;
+		}
+	}
+
+	fn key_before_prefix(mut prefix: Vec<u8>) -> Vec<u8> {
+		let last = prefix.iter_mut().last().unwrap();
+		*last = last.checked_sub(1).unwrap_or_else(|| unimplemented!());
```

Comment:
```suggestion
		*last -= 1;
```

Reply:
I still think it is better to have the mock function fail instead of silently return wrong result, so I put an assert instead and apply your suggestion